### PR TITLE
Added image search criteria

### PR DESCRIPTION
**Pull request by juskora**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-6620
| Versions      | master, 4.6
| Edition       | 

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/search/criteria_reference/imagefilesize_criterion.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+---
+
+# Image FileSize Criterion
+
+The `FileSize` Search Criterion searches for image with specified size.
+
+## Arguments
+
+- `min_size` - minimum file size expressed in MB
+- `max_size` - maximum file size expressed in MB
```
Comment (adamwojs) [inline docs/search/criteria_reference/imagefilesize_criterion.md:12]: This list is incomplete

File: docs/search/criteria_reference/imageorientation_criterion.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+description: Image Orientation Criterion
+---
+
+# Image Orientation Criterion
+
+The `Orientation` Search Criterion searches for image with specified orientation.
+
+## Arguments
+
+- `orientation` - string or array. Allowed orientation types: `portrait`, `landscape`, `square`.
```
Comment (adamwojs) [inline docs/search/criteria_reference/imageorientation_criterion.md:11]: This list is incomplete

File: docs/search/criteria_reference/imageorientation_criterion.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+## Example
+
+### PHP
+
+``` php
+$query = new MimeType(string $fieldDefIdentifier, $type);
```
Comment (adamwojs) [inline docs/search/criteria_reference/imageorientation_criterion.md:18]: Yet another invalid PHP code example

File: docs/search/criteria_reference/imagewidth_criterion.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+- `min_value` - minimum file size expressed in pixels
+- `max_value` - maximum file size expressed in pixels
+
+## Example
+
+### PHP
+
+``` php
+$query = new Width(string $fieldDefIdentifier, int $minValue = 0, ?int $maxValue = null);
```
Comment (adamwojs) [inline docs/search/criteria_reference/imagewidth_criterion.md:19]: Same as above

File: docs/search/criteria_reference/image_criterion.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
+---
+description: Image Criterion
+---
+
+# Image Criterion
+
+The `MimeType` Search Criterion searches for image by specified image attributes.
```
Comment (ciastektk) [inline docs/search/criteria_reference/image_criterion.md:7]:
```suggestion
The `Image` Search Criterion searches for image by specified image attributes.
```

File: docs/search/criteria_reference/imagefilesize_criterion.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        "fieldDefIdentifier": "image",
+        "size": {
+          "min": 0, 
+    "max": 1.5
+        }
+      }
+       }
+    }
+
+    ```
```
Comment (ciastektk) [inline docs/search/criteria_reference/imagefilesize_criterion.md:56]: Bad indentation

File: docs/search/criteria_reference/imageorientation_criterion.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
+---
+description: Image Orientation Criterion
+---
+
+# Image Orientation Criterion
+
+The `Orientation` Search Criterion searches for image with specified mime orientation(s).
```
Comment (ciastektk) [inline docs/search/criteria_reference/imageorientation_criterion.md:7]:
```suggestion
The `Orientation` Search Criterion searches for image with specified orientation(s).
```

File: docs/search/criteria_reference/imageorientation_criterion.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+=== "JSON"
+
+    ```json
+    "Query": {
+        "Filter": {
+            "ImageOrientationCriterion": {
+                "fieldDefIdentifier": "image",
+                "orientation": "landscape"
+    }
+       }
```
Comment (ciastektk) [inline docs/search/criteria_reference/imageorientation_criterion.md:57]:
```suggestion
            "ImageOrientationCriterion": {
                "fieldDefIdentifier": "image",
                "orientation": "landscape"
            }
        }
```

File: docs/search/criteria_reference/search_criteria_reference.md [hunk 9]
```diff
@@ -33,7 +33,13 @@ Due to this storage limitation, searching content using the Country Field Type o
...
 |[Field](field_criterion.md)|Content of one of content item's Fields|&#10004; |&#10004; | |
 |[FieldRelation](fieldrelation_criterion.md)|Content items the content in question has Relations to|&#10004; |&#10004; | |
-|[FullText](fulltext_criterion.md)|Full text content of a content item's Fields|&#10004; |&#10004; | |
+|[FullText](fulltext_criterion.md)|Full text content of a Content item's Fields|&#10004; |&#10004; | |
+|[ImageDimensions](imagedimensions_criterion.md)|Image dimensions: height and width||||
+|[ImageFileSize](imagefilesize_criterion.md)|Image size in MB||||
+|[ImageHeight](imageheight_criterion.md)|Image height in pixels|||
+|[ImageMimeType](imagemimetype_criterion.md)|Image type|||
+|[ImageOrientation](imageorientation_criterion.md)|Image orientation|||
+|[ImageWidth](imagewidth_criterion.md)|Image widht in pixels|||
```
Comment (ciastektk) [inline docs/search/criteria_reference/search_criteria_reference.md:42]: Missing reference to Image criterion.

Each image criterion supports Content and Location search.

File: docs/search/criteria_reference/imagedimensions_criterion.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+              <min>100</min>
+              <max>1000</max>
+            </width>
+            <height>
+              <min>500</min>
+              <max>1500</max>
+            </height>
+          </ImageDimensionsCriterion>
+        </Filter>
+    </Query>
```
Comment (ciastektk) [inline docs/search/criteria_reference/imagedimensions_criterion.md:54]: Bad indentation.

https://ez-systems-developer-documentation--2225.com.readthedocs.build/en/2225/search/criteria_reference/imagedimensions_criterion/

vs

https://ez-systems-developer-documentation--2225.com.readthedocs.build/en/2225/search/criteria_reference/imagefilesize_criterion/

**Review by ciastektk — CHANGES_REQUESTED** [review]

**Review by julitafalcondusza — APPROVED** [review]

File: docs/search/criteria_reference/search_criteria_reference.md [hunk 11]
```diff
@@ -33,7 +33,14 @@ Due to this storage limitation, searching content using the Country Field Type o
...
 |[FieldRelation](fieldrelation_criterion.md)|Content items the content in question has Relations to|&#10004; |&#10004; | |
-|[FullText](fulltext_criterion.md)|Full text content of a content item's Fields|&#10004; |&#10004; | |
+|[FullText](fulltext_criterion.md)|Full text content of a Content item's Fields|&#10004; |&#10004; | |
+|[Image](image_criterion.md)|Image by specified image attributes|&#10004;|&#10004;||
+|[ImageDimensions](imagedimensions_criterion.md)|Image dimensions: height and width|&#10004;|&#10004;||
+|[ImageFileSize](imagefilesize_criterion.md)|Image size in MB|&#10004;|&#10004;||
+|[ImageHeight](imageheight_criterion.md)|Image height in pixels|&#10004;|&#10004;||
+|[ImageMimeType](imagemimetype_criterion.md)|Image type|&#10004;|&#10004;||
+|[ImageOrientation](imageorientation_criterion.md)|Image orientation|&#10004;|&#10004;||
+|[ImageWidth](imagewidth_criterion.md)|Image widht in pixels|&#10004;|&#10004;||
```
Comment (dabrt) [inline docs/search/criteria_reference/search_criteria_reference.md:43]:
```suggestion
|[ImageWidth](imagewidth_criterion.md)|Image width in pixels|&#10004;|&#10004;||
```

File: docs/search/criteria_reference/search_criteria_reference.md [hunk 12]
```diff
@@ -33,7 +33,14 @@ Due to this storage limitation, searching content using the Country Field Type o
 |[Depth](depth_criterion.md)|Location depth in the Content tree| |&#10004; |&#10004; |
 |[Field](field_criterion.md)|Content of one of content item's Fields|&#10004; |&#10004; | |
 |[FieldRelation](fieldrelation_criterion.md)|Content items the content in question has Relations to|&#10004; |&#10004; | |
-|[FullText](fulltext_criterion.md)|Full text content of a content item's Fields|&#10004; |&#10004; | |
+|[FullText](fulltext_criterion.md)|Full text content of a Content item's Fields|&#10004; |&#10004; | |
```
Comment (dabrt) [inline docs/search/criteria_reference/search_criteria_reference.md:36]:
```suggestion
|[FullText](fulltext_criterion.md)|Full text content of a content item's Fields|&#10004; |&#10004; | |
```

File: docs/search/criteria_reference/image_criterion.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+
+- `fieldDefIdentifier` - string representing the identifier of the Field
+- `imageCriteriaData` - array representing image attributes. All attributes are optional.
+
+## Example
+
+### PHP
+
+``` php
+
```
Comment (dabrt) [inline docs/search/criteria_reference/image_criterion.md:19]:
```suggestion
```

File: docs/search/criteria_reference/imagedimensions_criterion.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+- `fieldDefIdentifier` - string representing the identifier of the Field
+- `imageCriteriaData` - an array representing minimum and maximum values for width and height, expressed in pixels
+
+## Example
+
+### PHP
+
+``` php
+
```
Comment (dabrt) [inline docs/search/criteria_reference/imagedimensions_criterion.md:19]:
```suggestion
```

File: docs/search/criteria_reference/imagedimensions_criterion.md [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+                    <max>1000</max>
+                </width>
+                <height>
+                    <min>500</min>
+                    <max>1500</max>
+                </height>
+            </ImageDimensionsCriterion>
+        </Filter>
+    </Query>
+
```
Comment (dabrt) [inline docs/search/criteria_reference/imagedimensions_criterion.md:55]:
```suggestion
    ```

File: docs/search/criteria_reference/imagedimensions_criterion.md [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+        'max' => 1000, // (default: null, optional)
+    ],
+    'height' => [
+        'min' => 500, // (default: 0, optional)
+        'max' => 1500, // (default: null, optional)
+    ],
+];
+
+$query->query = new Criterion\Dimensions('image', $imageCriteriaData);
+
```
Comment (dabrt) [inline docs/search/criteria_reference/imagedimensions_criterion.md:32]:
```suggestion
```

File: docs/search/criteria_reference/imagefilesize_criterion.md [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+            "ImageFileSizeCriterion":{
+                "fieldDefIdentifier": "image",
+                "size": {
+                    "min": 0, 
+                    "max": 1.5
+                }
+            }
+        }
+    }
+
```
Comment (dabrt) [inline docs/search/criteria_reference/imagefilesize_criterion.md:55]:
```suggestion
```

File: docs/search/criteria_reference/imagemimetype_criterion.md [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+or 
+
+```php
+$mimeTypes = [
+    'image/jpeg',
+    'image/png',
+];
+
+$query->query = new Criterion\MimeType('image', $mimeTypes);
+
```
Comment (dabrt) [inline docs/search/criteria_reference/imagemimetype_criterion.md:31]:
```suggestion
```

File: docs/search/criteria_reference/imageorientation_criterion.md [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+
+OR
+
+$orientations = [
+    'landscape',
+    'portrait',
+];
+
+$query->query = new Criterion\Orientation('image', $orientations);
+
```
Comment (dabrt) [inline docs/search/criteria_reference/imageorientation_criterion.md:30]:
```suggestion
```

**Review by dabrt — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

**Pull request merged**